utils/pretty.py

```diff
@@ -24,13 +24,55 @@ def x(o):
     return x
 
 
+def process_attrs(attrs):
+    """Enables us for more freedom in attribute specification"""
+    if isinstance(attrs, basestring):
+        return [attrs]
+    elif attrs is None:
+        return []
+    elif isinstance(attrs, (list, tuple)):
+        return attrs
+    else:
+        raise TypeError(
+            "pretty_attrs should be tuple, list, str or None, not {}".format(type(attrs).__name__))
+
+
 class Pretty(object):
-    """A mixin that prints repr as <MyClass field1=..., field2=...>. The
-       fields that will be printed should be stored in the class's
-       pretty_attrs attribute (none by default).
+    """A mixin that prints repr as <MyClass field1=..., field2=...>. Provides default constructor
 
+    The fields that will be printed should be stored in the class's pretty_attrs attribute
+    (none by default). It also provides a default constructor for the values specified.
+
+    the ``pretty_attrs`` can be list, tuple, str or None. The ``required_attrs`` is a list of
+    attributes that are required for a successful construction of the object, same structure as
+    pretty_attrs. ``default_attrs`` is a dictionary of default values. Keys must exist in
+    ``pretty_attrs`` for it to work.
     """
-    pretty_attrs = []
+    pretty_attrs = None
+    required_attrs = None
+    default_attrs = {}
+
+    def __init__(self, *args, **kwargs):
+        pretty_attrs = process_attrs(type(self).pretty_attrs)
```

Comment:
Seems like process_attrs would benefit from being a classmethod (or being wrapped by one), since it's always being called (or could always be called) with `type(self)`.